tasks/broad/BamProcessing.wdl

```diff
@@ -22,12 +22,16 @@ task SortSam {
     String output_bam_basename
     Int preemptible_tries
     Int compression_level
+    Int memory_multiplier = 1
   }
   # SortSam spills to disk a lot more because we are only store 300000 records in RAM now because its faster for our data so it needs
   # more disk space.  Also it spills to disk in an uncompressed format so we need to account for that with a larger multiplier
   Float sort_sam_disk_multiplier = 3.25
   Int disk_size = ceil(sort_sam_disk_multiplier * size(input_bam, "GiB")) + 20
 
+  Int machine_mem_mb = ceil(5000 * memory_multiplier)
+
+
   command {
     java -Dsamjdk.compression_level=~{compression_level} -Xms4000m -Xmx4500m -jar /usr/picard/picard.jar \
```

Comment:
You'll need to scale the -Xms and -Xmx parameters accordingly. I'm sure we have examples of that too.  Typically would set Xmx about 500 less than machine_mem_mb and Xms 1000 less.